FirebaseDatabase/Sources/Core/FPersistentConnection.m

```diff
@@ -759,6 +811,45 @@ - (void)putInternal:(id)data
     }
 }
 
+- (void)get:(NSString *)pathString
+      withParams:(NSDictionary *)queryWireProtocolParams
+    withCallback:(fbt_void_nsstring_id_nsstring)onComplete {
+    NSMutableDictionary *request = [NSMutableDictionary
+        dictionaryWithObjectsAndKeys:pathString, kFWPRequestPath,
+                                     queryWireProtocolParams,
+                                     kFWPRequestQueries, nil];
+    FOutstandingGet *get = [[FOutstandingGet alloc] init];
+    get.request = request;
+    get.onCompleteBlock = onComplete;
+    get.sent = NO;
+
+    NSNumber *index = [self.getCounter getAndIncrement];
+    self.outstandingGets[index] = get;
+
+    if (![self connected]) {
+        dispatch_after(
+            dispatch_time(DISPATCH_TIME_NOW, kPersistentConnGetConnectTimeout),
+            self.dispatchQueue, ^{
+              FOutstandingGet *get = self.outstandingGets[index];
```

Comment:
This is missing the log statement from Android, which would also improve clarity here.